File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/actions/fight/GameActionFightCastRequestMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameActionFightCastRequestMessage extends MessageDeserializer {
  val messageId = 1005

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameActionFightCastRequestMessage.messageId

    val spellId: Short
    val cellId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(spellId)
      buf.writeShort(cellId)
    }

  }

  protected def create(spellId: Short, cellId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val spellId = buf.readShort()
    val cellId = buf.readShort()

    create(spellId, cellId)
  }

}

object GameActionFightCastRequestMessage extends GameActionFightCastRequestMessage {
  case class MessageImpl(spellId: Short, cellId: Short) extends Message
  type Target = MessageImpl

  override def create(spellId: Short, cellId: Short) = MessageImpl(spellId, cellId)

  def apply(spellId: Short, cellId: Short) = create(spellId, cellId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameActionFightCastRequestMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameActionFightCastRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 1005;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short spellId;
    public short cellId;
    
    public GameActionFightCastRequestMessage() { }
    
    public GameActionFightCastRequestMessage(short spellId, short cellId) {
        this.spellId = spellId;
        this.cellId = cellId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(spellId);
        buf.writeShort(cellId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        spellId = buf.readShort();
        if (spellId < 0)
            throw new RuntimeException("Forbidden value on spellId = " + spellId + ", it doesn't respect the following condition : spellId < 0");
        cellId = buf.readShort();
        if (cellId < -1 || cellId > 559)
            throw new RuntimeException("Forbidden value on cellId = " + cellId + ", it doesn't respect the following condition : cellId < -1 || cellId > 559");
    }
    
}
